client/server/pages/index.js

```diff
@@ -920,6 +920,28 @@ export default function pages() {
 
 	app.set( 'views', __dirname );
 
+	// Limit allowed header sizes
+	app.use( ( req, res, next ) => {
+		const MAX_HEADER_LENGTH = 8192; // 8 KB total for all headers
+		const MAX_USER_AGENT_LENGTH = 1024; // 1 KB for User-Agent header
```

Comment:
It may be worth it to hoist these to the top of the module. Since this is server code and these variables will stay in memory across sessions. This middleware will run thousands of times per minute.